entity-server-api/src/main/java/org/terracotta/entity/CommonServerEntity.java

```diff
@@ -46,4 +46,12 @@
    *  respect to all other messages enqueued for the entity.</p>
    */
   void destroy();
+
+  /**
+   * Notify an entity that a particular client has a shared understanding of the state
+   * of the system, such as after the client disappears, or after reconnect has finished.
+   *
+   * @param client client
+   */
+  void notifyClientIsSynched(ClientDescriptor client);
```

Comment:
Yeah. It wouldbe incumbent on the server to ensure it is called in a method that guarantees it is useful/reliable. Disconnect for sure is easy, because once the client is gone, we don't need the history. reconnect may be tougher, we will see. There may also be other cases where it could be done.